internal/metrics/client_test.go

```
@@ -0,0 +1,665 @@
+		expectedError       string
+	}{
+		{
+			name: "successful RPS query",
+			metrics: []parser.Metric{
+				{Name: "rps"},
+			},
+			mockResponse: QueryResponse{
+				Status: "success",
+				Data: struct {
+					ResultType string `json:"resultType"`
+					Result     []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					} `json:"result"`
+				}{
+					ResultType: "vector",
+					Result: []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					}{
+						{
+							Value: []interface{}{1234567890, "42.5"},
+						},
+					},
+				},
+			},
+			mockStatus:          http.StatusOK,
+			expectedMetricKey:   "rps",
+			expectedMetricValue: 43, // Rounded up from 42.5
+		},
+		{
+			name: "custom metric",
+			metrics: []parser.Metric{
+				{Name: "custom_metric"},
+			},
+			mockResponse: QueryResponse{
+				Status: "success",
+				Data: struct {
+					ResultType string `json:"resultType"`
+					Result     []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					} `json:"result"`
+				}{
+					ResultType: "vector",
+					Result: []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					}{
+						{
+							Value: []interface{}{1234567890, "42.5"},
+						},
+					},
+				},
+			},
+			mockStatus:          http.StatusOK,
+			expectedMetricKey:   "custom_metric",
+			expectedMetricValue: 43, // Rounded up from 42.5
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tt.mockStatus)
+				json.NewEncoder(w).Encode(tt.mockResponse)
```

You're calling this method a few times in these tests but the error is never checked. I suggest using `require.NoError`.